### ISSUE TITLE
register without email

### ISSUE THREAD

**Issue by streaps** [xenolf/lego#277]
Can lego register an account without an email address?

It is possible with other tools (letsencrypt.sh) and I don't have any need to share my email address with letsencrypt. see https://community.letsencrypt.org/t/email-address-disclosures-june-11-2016/17025

**Comment by mholt** [comment]
The underlying acme package can, but the command doesn't currently allow that.

**Comment by hairyhenderson** [comment]
@mholt is there any reason why `lego` _shouldn't_ allow registering without an e-mail address?

**Comment by mholt** [comment]
Other than it being strongly, strongly discouraged? Not that I know of, no. @xenolf would know better though.

**Comment by hairyhenderson** [comment]
Thanks @mholt!

**Comment by xenolf** [comment]
Yeah like @mholt said, there is no other reason then it being discouraged. I'm not sure why you would like to specifically not have a recoverable account.

**Comment by hairyhenderson** [comment]
@xenolf It's not a _strong_ use-case, to be clear - more of a vague idea. My main use-case is for developers who set up temporary sites that usually live for 2-3 days (7 tops), and it'd be _slightly_ inconvenient for them to have to specify an e-mail address when there's never really a need to renew. Right now that scenario involves self-signed certificates, which are increasingly annoying to deal with.

Mostly I'm not 100% clear on exactly what LE/ACME _does_ with the e-mail address - I'll do some more reading and figure it out.

**Comment by smtalk** [comment]
@hairyhenderson similar needs here, we just don't know real email addresses on the systems we're auto-deploying the certs. On the ones we 'known' them - some certs are just temporary ones or different sets, and they get flooded with expiring certificate notices then (they're confusing, as they include a different set of SANs). Having ability not to specify the email would be really great!

**Comment by smtalk** [comment]
@ldez just wondering if there is a chance to see this included into the next release? We'd replace our own LE client with lego on all the installations of DirectAdmin, that's the only thing left.. :)

**Comment by rafaelgieschke** [comment]
I tried to implement this in <https://github.com/go-acme/lego/issues/277>.

**Comment by ldez** [comment]
I don't really understand the need:
- if you are a company, it's highly recommended to use an email, like that if you have a rate limit problem Let's Encrypt will be able to increase your limit for this email
- you can use an email alias to catch all those emails, and "hide" them
- Let's Encrypt discouraged to do that

I don't want to change the default behavior, that promotes the use of an email like Let's Encrypt recommends, without a strong use case.

For now, I don't see any strong use cases but if you have one, please share it.

**Comment by rafaelgieschke** [comment]
I see the same use case as <https://github.com/go-acme/lego/issues/277#issuecomment-287442620>. You might not have an (appropriate) email address or you might not want to give it to your ACME provider. You might not want to give up control of your account to your email provider. You might want to have distinct accounts for different domains not connected by any email address.

Currently, it encourages bad behavior for this use case. Let's Encrypt only accepts email addresses with domains on the Public Suffix List, so you are encouraged to make up something like `absolutelydontdothis@no-email.com`, `$random@example.com`, `no-email@yourrealdomain.example`, or `invalid@0.0.0.0.in-addr.arpa` (and you never have to verify any email address before issuing certificates), which is far worse for both Let's Encrypt and security than specifying no email address at all (and accepting that Let's Encrypt cannot contact you, e.g., when revoking certificates and that your account might be lost). You might also be encouraged to use other tools supporting registration without an email, like mentioned in <https://github.com/go-acme/lego/issues/277#issue-172652825>.

From an implementation point of view, the current behavior (with `userID: email,`, `// TODO: move to account struct? Currently MUST pass email.`) feels more unpolished than (optionally) decoupling userID and email.

I do see your point, though, that people might get themselves into trouble and regret not specifying an email address later on but maybe a strong discouragement in `--user-id`'s help text and forcing users to set an option with a deterring name (`--i-accept-dangers-from-no-email-and-might-miss-crucial-information-about-my-certificates-and-my-account-might-be-lost`) might work?

**Comment by tisba** [comment]
Just found this issue as I was wondering why lego requires an email. I'm genuinely interested in understanding and not trying to offend anyone.

[@xenolf said](https://github.com/go-acme/lego/issues/277#issuecomment-287439505)

> Yeah like @mholt said, there is no other reason then it being discouraged. I'm not sure why you would like to specifically not have a recoverable account.

That might be relevant for some, but certainly not for the majority. Unless I'm overlooking something, the email is only used for

1.  [expiration email](https://letsencrypt.org/docs/expiration-emails/) (as a hobbyist I don't care, as a professional I have monitoring)
2. [certificate revocation](https://letsencrypt.org/docs/revoking/) (you can always do that if you can prove control)
3. General communication from Let's Encrypt, security infos etc (IMO, same argument as with 1.)
4. [rate limits](https://letsencrypt.org/docs/rate-limits/)

Only rate limits might be a reason to have a "fixed" and "recoverable" account, so that you can request an override if you are a large provider.

I don't quite follow what else could be meant by "recoverable account" and why Let's Encrypt discourages to leave out the mail. Do you have a reference for that, @ldez? (https://github.com/go-acme/lego/issues/277#issuecomment-806288015). I can't find anything in that regard in the Let's Encrypt documentation, nor in [EFF's certbot docs](https://eff-certbot.readthedocs.io/en/stable/using.html). AFAIK certbot is the de-facto recommended tool and is quite clearly communicating email to be optional.

**Comment by tisba** [comment]
Thank you, that was the missing piece.

From your comment I take that by know it would indeed be acceptable to make email optional for lego? If that's the case, I'd take a look if I can contribute a PR to change this requirement.

**Comment by mcpherrinm** [comment]
It should be possible to register without an email, because otherwise people will put in bogus values, which helps nobody.

It is, however, still valuable for most people to provide an email for communication about their certificates, so an explicit opt-out flag (like --no-email) is IMO the best approach.

**Comment by ldez** [comment]
> otherwise people will put in bogus values, which helps nobody and causes us to send emails into the void.

ok, this argument is strong enough for me, so @mcpherrinm you can open a PR.

**Comment by asyslinux** [comment]
.....I don’t understand why we can’t finally make this stinking --no-email flag? :)

**Comment by tisba** [comment]
jFYI, as you might have noticed, Let's Encrypt just ended their expiration notification service June 4, 2025: https://letsencrypt.org/2025/01/22/ending-expiration-emails.
